trie/database.go

```diff
@@ -18,11 +18,75 @@ package trie
 
 import (
 	"github.com/ChainSafe/gossamer/polkadb"
-	//"sync"
+	"sync"
 )
 
 // Database is a wrapper around a polkadb
 type Database struct {
-	db polkadb.Database
-	//lock sync.RWMutex
+	db     polkadb.Database
+	batch  polkadb.Batch
+	lock   sync.RWMutex
+	hasher *Hasher
+}
+
+// WriteToDB writes the trie to the underlying database batch writer
+// Stores the merkle value of the node as the key and the encoded node as the value
+// This does not actually write to the db, just to the batch writer
+// Commit must be called afterwards to finish writing to the db
+func (t *Trie) WriteToDB() error {
+	t.db.batch = t.db.db.NewBatch()
+	return t.writeToDB(t.root)
+}
+
+// writeToDB recursively attempts to write each node in the trie to the db batch writer
+func (t *Trie) writeToDB(n node) error {
+	_, err := t.writeNodeToDB(n)
+	if err != nil {
+		return err
+	}
+
+	switch n := n.(type) {
```

Comment:
Do we need a default statement here or is there an implicit fallthrough if `n != *branch` case?

Reply:
there is an implicit fallthrough